src/plugins/management/public/components/feature_cards/feature_cards.tsx

```diff
@@ -30,26 +30,20 @@ export const FeatureCards = ({
   pageDescription,
   navigationUI: { HeaderControl },
 }: FeatureCardsProps) => {
-  const itemsPerRow = 4;
   const groupedCardForDisplay = useMemo(() => {
-    const grouped: Array<{ category?: AppCategory; navLinks: ChromeNavLink[][] }> = [];
+    const grouped: Array<{ category?: AppCategory; navLinks: ChromeNavLink[] }> = [];
     // The navLinks has already been sorted based on link / category's order,
     // so it is safe to group the links here.
     navLinks.forEach((link) => {
       let lastGroup = grouped.length ? grouped[grouped.length - 1] : undefined;
       if (!lastGroup || lastGroup.category?.id !== link.category?.id) {
-        lastGroup = { category: link.category, navLinks: [[]] };
+        lastGroup = { category: link.category, navLinks: [] };
         grouped.push(lastGroup);
       }
-      const lastRow = lastGroup.navLinks[lastGroup.navLinks.length - 1];
-      if (lastRow.length < itemsPerRow) {
-        lastRow.push(link);
-      } else {
-        lastGroup.navLinks.push([link]);
-      }
+      lastGroup.navLinks.push(link);
```

Comment:
Why risk it? It would be so much cheaper to just make a Map based on `link.category?.id` and group them than check the length of an array so many times.

Reply:
Thanks, updated. Actually we do not need to get last group in every loop.